src/terrain/terrain_base.hpp

```diff
@@ -170,9 +285,11 @@ class TerrainBase : public voxel_utility::VoxelBase {
      * @param xyz vector index
      * @return const std::array<int, 3> position in space
      */
-    inline const std::array<int, 3>
-    sop(int xyz) const {
-        return {xyz / (Y_MAX * Z_MAX), (xyz / Z_MAX) % Y_MAX, xyz % (Z_MAX)};
+    inline const std::array<uint16_t, 3>
+    sop(uint32_t xyz) const {
+        return {
+            uint16_t(xyz / (Y_MAX * Z_MAX)), uint16_t((xyz / Z_MAX) % Y_MAX),
+            uint16_t(xyz % (Z_MAX))};
     }
```

Comment:
I worry about overflow here

Reply:
no, because the values should be stored as `uint16_t` in each tile.